src/app/screens/do-routine/routine-runner/index.tsx

```diff
@@ -25,23 +32,26 @@ import {
   StartBadge,
   StyledBottomSheet,
 } from './styles';
-import { Props } from './types';
+import { ExerciseStatus, Props } from './types';
 
-export const RoutineRunnerScreen: FC<Props> = ({ route, navigation }) => {
-  const { routine } = route.params;
+export const RoutineRunnerScreen: FC<Props> = ({ navigation }) => {
   const { t } = useTranslation();
   const theme = useTheme();
+  const dispatch = useAppDispatch();
   const { isPaused, start, pause, formattedTime } = useTimer();
-  const [routineExercisesData, setRoutineExercisesData] = useState<
-    RoutineExercise[] | []
-  >([]);
-  const [currentExerciseId, setCurrentExerciseId] = useState<number | null>(
-    null,
+  const { activeRoutine, currentExercise, summaryRoutine } = useAppSelector(
+    state => state.routine,
   );
 
+  const exercises = useMemo<RoutineExercise[]>(() => {
+    return activeRoutine?.routine.exercises || [];
+  }, [activeRoutine?.routine.exercises]);
```

Comment:
You could try something like this to improve a bit more the code
```suggestion
  const exercises = useMemo<RoutineExercise[]>(() => 
    (activeRoutine?.routine.exercises || [])
  , [activeRoutine?.routine.exercises]);
```